community/php7/APKBUILD

```diff
@@ -157,203 +322,285 @@ build() {
 		--enable-phpdbg \
 		--enable-phpdbg-webhelper \
 		--disable-cgi \
-		--disable-cli \
-		|| return 1
+		--disable-cli
+
 	# apache2 module
 	_build --disable-phpdbg \
 		--disable-cgi \
 		--disable-cli \
-		--with-apxs2 \
-		|| return 1
-	mv libs/libphp7.so sapi/apache2handler/mod_php7.so
-	# cgi,cli,fpm,embed,litespeed
+		--with-apxs2
+	mv libs/libphp$_suffix.so sapi/apache2handler/mod_php$_suffix.so
+
+	# cgi, cli, fpm, embed, litespeed
 	_build --disable-phpdbg \
 		--enable-fpm \
 		--enable-embed \
-		--with-litespeed \
-		|| return 1
+		--with-litespeed
+}
+
+check() {
+	cd "$builddir"
+
+	# PHP is so stupid that it's not able to resolve dependencies
+	# between extensions and load them in correct order, so we must
+	# help it...
+	# * opcache is Zend extension, it's handled specially in Makefile
+	# * imap can't be loaded together with recode, so we must omit it
+	local php_modules=$(_extensions_by_deps_order \
+		| grep -vx opcache \
+		| grep -vx imap \
+		| xargs -n 1 printf "'$builddir/modules/%s.la' ")
+	sed -i "/^PHP_TEST_SHARED_EXTENSIONS/,/extension=/ \
+		s|in \$(PHP_MODULES)\"*|in $php_modules|" Makefile
+
+	NO_INTERACTION=1 REPORT_EXIT_STATUS=1 \
+		SKIP_SLOW_TESTS=1 SKIP_ONLINE_TESTS=1 TEST_TIMEOUT=10 \
+		TZ= LANG= LC_ALL= \
+		make test
+
+	echo 'NOTE: We have skipped quite a lot tests, see disabled-tests.list.'
 }
 
 package() {
 	cd "$builddir"
 
-	make -j1 INSTALL_ROOT="$pkgdir" install || return 1
+	make -j1 INSTALL_ROOT="$pkgdir" install
 
-	install -Dm644 php.ini-production "$pkgdir"/etc/$pkgname/php.ini || return 1
+	install -Dm644 php.ini-production "$pkgdir"/etc/$pkgname/php.ini
 
 	local file; for file in pear peardev pecl; do
-		sed -i -e "s~/usr/bin/php~/usr/bin/php7~g" \
-			-e "s~PHP=php~PHP=php7~" \
-			"$pkgdir"/usr/bin/$file || return 1
+		sed -i -e "s|/usr/bin/php|/usr/bin/php$_suffix|g" \
+			-e "s|PHP=php|PHP=php$_suffix|" \
+			"$pkgdir"/usr/bin/$file
 	done
-	paxmark -m "$pkgdir"/usr/bin/php7
 
-	find "$pkgdir" -name '.*' | xargs rm -rf || return 1
+	find "$pkgdir" -name '.*' | xargs rm -rf
 	rmdir "$pkgdir"/var/run
+
+	[ "$_default_php" = yes ] \
+		&& ln -s php$_suffix "$pkgdir"/usr/bin/php
 }
 
 dev() {
 	default_dev
 	replaces="php-dev"
-	depends="$depends $pkgname $pkgname-pear"
```

Comment:
Why you not use global depends_dev instead?
And dev does not depend on pear.

For all 3rd party php-extensions to build need php (cli), autoconf and most of them pcre-dev, so look in my variant:
depends_dev="$pkgname autoconf pcre-dev"

Also I split pear on two:  pear (just runtime) and pecl (same as pear but with ability to build extension from source)

pecl() {
        replaces=
        pkgdesc="$pkgdesc (php extension community library)"
        depends="$pkgname-pear $pkgname-dev gcc make musl-dev"
        mkdir -p "$subpkgdir"
}